api/resolvers/user.js

```diff
@@ -541,15 +541,22 @@ export default {
         throw new GraphQLError('you are not allowed to generate api keys', { extensions: { code: 'FORBIDDEN' } })
       }
 
-      const [{ apiKey }] = await models.$queryRaw`UPDATE users SET "apiKey" = encode(gen_random_bytes(32), 'base64')::CHAR(32) WHERE id = ${me.id} RETURNING "apiKey"`
+      // I trust postgres CSPRNG more than the one from JS
+      const [{ apiKey, apiKeyHash }] = await models.$queryRaw`
+      SELECT "apiKey", encode(digest("apiKey", 'sha256'), 'hex') AS "apiKeyHash"
+      FROM (
+        SELECT encode(gen_random_bytes(32), 'base64')::CHAR(32) as "apiKey"
+      ) rng`
```

Comment:
what does `rng` do here?

Reply:
It's the alias for the subquery and I picked "rng" for "random number generator":

```
stackernews=# SELECT "apiKey", encode(digest("apiKey", 'sha256'), 'hex') AS "apiKeyHash"
stackernews-#       FROM (
stackernews(#         SELECT encode(gen_random_bytes(32), 'base64')::CHAR(32) as "apiKey"
stackernews(#       );
ERROR:  subquery in FROM must have an alias
LINE 2:       FROM (
                   ^
HINT:  For example, FROM (SELECT ...) [AS] foo.
```

Reply:
Ah ok I wasn't thinking the alias was required. Thanks!